src/routes/safe/components/Transactions/helpers/TxModalWrapper/index.tsx

```diff
@@ -238,11 +244,12 @@ export const TxModalWrapper = ({
 
   return (
     <EditableTxParameters
+      key={manualGasLimit || gasLimit}
       isExecution={doExecute}
-      ethGasLimit={gasLimit}
-      ethGasPrice={gasPriceFormatted}
-      ethMaxPrioFee={gasMaxPrioFeeFormatted}
-      safeTxGas={safeTxGas}
+      ethGasLimit={manualGasLimit || gasLimit}
+      ethGasPrice={manualGasPrice || gasPriceFormatted}
+      ethMaxPrioFee={manualMaxPrioFee || gasMaxPrioFeeFormatted}
+      safeTxGas={manualSafeTxGas || safeTxGas || safeTxGasEstimation}
```

Comment:
I'm not sure if we should leave this form as it was.

IMHO if the user changes _only_ the safeTxGas, it's supposed to re-estimate gas limit.

But one could also say that if they change any gas param, they should all be taken from the manually edited values.

Reply:
Yes, I would too. Let's keep these changes then. Not a fan of the `key` hack, but I hope we can do it better in web-core.

Reply:
Afaik the values we pass here are only used as the initial value of the form fields and are then managed by another state. Once the user changes any of the values it is properly reflected in the form. `manualX` is only being used for the estimation in this component.